### ISSUE TITLE
ValueError: None is only supported in the 1st dimension. Tensor 'image_tensor' has invalid shape '[None, None, None, 3]'.

### ISSUE THREAD

**Issue by Ringhu** [tensorflow/tensorflow#38558]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): pip install 
- TensorFlow version (or github SHA if from source): 2.1.0

**Command used to run the converter or code if you’re using the Python API**
If possible, please share a link to Colab/Jupyter/any notebook.

```
tflite_convert --saved_model_dir=/path/to/automl/efficientdet/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/saved_model/ --output_file=/path/to/automl/efficientdet/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/
```

**The output from the converter invocation**

```
Traceback (most recent call last):

  File "/home/hulining/.local/bin/tflite_convert", line 11, in <module>
    sys.exit(main())

  File "/home/hulining/.local/lib/python3.6/site-packages/tensorflow/lite/python/tflite_convert.py", line 503, in main

    app.run(main=run_main, argv=sys.argv[:1])

  File "/home/hulining/.local/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 40, in run

    _run(main=main, argv=argv, flags_parser=_parse_flags_tolerate_undef)

  File "/home/hulining/.local/lib/python3.6/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)

  File "/home/hulining/.local/lib/python3.6/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))

  File "/home/hulining/.local/lib/python3.6/site-packages/tensorflow/lite/python/tflite_convert.py", line 499, in run_main

    _convert_tf1_model(tflite_flags)

  File "/home/hulining/.local/lib/python3.6/site-packages/tensorflow/lite/python/tflite_convert.py", line 193, in _convert_tf1_model

    output_data = converter.convert()

  File "/home/hulining/.local/lib/python3.6/site-packages/tensorflow/lite/python/lite.py", line 811, in convert

    _get_tensor_name(tensor), shape_list))

ValueError: None is only supported in the 1st dimension. Tensor 'image_tensor' has invalid shape '[None, None, None, 3]'.

```

**Also, please include a link to the saved model or GraphDef**

```
https://github.com/google/automl/tree/master/efficientdet

http://download.tensorflow.org/models/object_detection/ssd_resnet50_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03.tar.gz

http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03.tar.gz
```

**Failure details**
I tried  several ways like Python API and command line and several models above to convert a .pb file to a tflite file. Both of them failed with the error above.
don't know how to slove the error.

**Comment by jvishnuvardhan** [comment]
@Ringhu Can you please share a standalone code to reproduce the issue? Thanks!

**Comment by Ringhu** [comment]
Of course. @jvishnuvardhan 
with the command line mode, I used:
'''
 tflite_convert --saved_model_dir=/path/to/automl/efficientdet/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/saved_model/ --output_file=/path/to/automl/efficientdet/ssd_mobilenet_v1_fpn_shared_box_predictor_640x640_coco14_sync_2018_07_03/
'''
with the python API code, the code is:
'''
converter = tf.lite.TFLiteConverter.from_saved_model('saved_model',signature_keys=['serving_default'])

converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.experimental_new_converter = True
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS, tf.lite.OpsSet.SELECT_TF_OPS]

tflite_model = converter.convert()

open("saved_model/converted_model.tflite", "wb").write(tflite_model)
'''
The transfered model may not be same (all from the link I shared above), but the error is the same.

BTW, my GPU is RTX 2080 Ti as another reference, don't know if it will affect.

**Comment by mgalgs** [comment]
Same issue here, one more example program and stack trace for reference:

<details>
<summary>
convert_to_tflite.py
</summary>

```python
import sys
import os

import tensorflow as tf


def usage():
    proggie = os.path.basename(sys.argv[0])
    print(f"""\
Usage: {proggie} <input_saved_model_dir> <output>

Example: {proggie} output_inference_graph_v1.pb/saved_model converted_model.tflite""")


if __name__ == "__main__":
    if len(sys.argv) != 3 or '-h' in sys.argv or '--help' in sys.argv:
        usage()
        sys.exit(1)

    saved_model_dir, outfile = sys.argv[1:]
    converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
    tflite_model = converter.convert()
    with open(outfile, "wb") as f:
        f.write(tflite_model)
    print("f{outfile} written")
```
</details>

<details>
<summary>output</summary>

```
$ docker exec tf python /tf/notebooks/training/routesv1/convert_to_tflite.py /tf/notebooks/training/routesv1/trained-inference-graphs/output_inference_graph_v1.pb/saved_model /tf/notebooks/training/routesv1/trained-inference-graphs/converted_model.tflite
2020-04-16 05:41:49.534053: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 AVX512F FMA
2020-04-16 05:41:49.565995: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3301490000 Hz
2020-04-16 05:41:49.567745: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x545a200 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-04-16 05:41:49.567786: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
WARNING:tensorflow:From /usr/local/lib/python3.6/dist-packages/tensorflow_core/lite/python/convert_saved_model.py:60: load (from tensorflow.python.saved_model.loader_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.loader.load or tf.compat.v1.saved_model.load. There will be a new function for importing SavedModels in Tensorflow 2.0.
2020-04-16 05:41:51.050535: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2020-04-16 05:41:51.050690: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2020-04-16 05:41:51.178923: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:786] Optimization results for grappler item: graph_to_optimize
2020-04-16 05:41:51.179140: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:788]   function_optimizer: function_optimizer did nothing. time = 0.354ms.
2020-04-16 05:41:51.179144: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:788]   function_optimizer: function_optimizer did nothing. time = 0.001ms.
Traceback (most recent call last):
  File "/tf/notebooks/training/routesv1/convert_to_tflite.py", line 22, in <module>
    tflite_model = converter.convert()
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/lite/python/lite.py", line 896, in convert
    _get_tensor_name(tensor), shape_list))
ValueError: None is only supported in the 1st dimension. Tensor 'image_tensor' has invalid shape '[None, None, None, 3]'.
```
</details>

<details>
<summary>nvidia-smi</summary>

```
$ nvidia-smi
Wed Apr 15 22:43:06 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.82       Driver Version: 440.82       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1070    Off  | 00000000:65:00.0  On |                  N/A |
|  0%   48C    P8    14W / 166W |    477MiB /  8118MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0   1332156      G   /usr/lib/Xorg                                212MiB |
|    0   1332206      G   /usr/bin/gnome-shell                         172MiB |
|    0   1332729      G   alacritty                                     11MiB |
|    0   1857817      G   ...gs/.local/share/Steam/ubuntu12_32/steam    20MiB |
|    0   1857987      G   ./steamwebhelper                               3MiB |
|    0   1858032      G   ...vice-request-channel-token=355490097884    49MiB |
+-----------------------------------------------------------------------------+
```
</details>

(guessing the gpu isn't used for the conversion, but there ya go)

<details>
<summary>pip show tensorflow</summary>

```
$ docker exec tf pip show tensorflow
WARNING: The directory '/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Name: tensorflow
Version: 1.15.2
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /usr/local/lib/python3.6/dist-packages
Requires: grpcio, opt-einsum, astor, absl-py, keras-applications, gast, keras-preprocessing, termcolor, six, tensorboard, tensorflow-estimator, numpy, protobuf, wheel, wrapt, google-pasta
Required-by:
```
</details>

**Comment by bgy-proprietor** [comment]
I download yolov3 model , then covert to keras h5 file 
when to covert h5 to tflite ,has the same error 

''None is only supported in the 1st dimension"

**Comment by rgbarishian** [comment]
I am getting the same error converting from a Keras model to tflite. I followed this tutorial
https://stackoverflow.com/a/60136583/11575257
When importing my saved model I printed a summary:

Layer (type)                 Output Shape              Param #   
=================================================================
input_2 (InputLayer)         [(None, None, None, 3)]   0         
_________________________________________________________________
conv1_pad                        (None, None, None, 3)     0         
_________________________________________________________________
conv1                                (None, None, None, 32)    864      
...

However, before training if I print a summary, I get:
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         [(None, 224, 224, 3)]     0         
_________________________________________________________________
conv1_pad                       (None, 225, 225, 3)       0         
_________________________________________________________________
conv1                               (None, 112, 112, 32)      864       
...

Which has correct output shapes.

I believe that this shows that the model did not save properly and output shapes were not saved in the model.

**Comment by jvishnuvardhan** [comment]
@Ringhu Sorry for the delay in my response. I tried converting the `mobilenet` model and was successful. Can you please check the [gist](https://colab.research.google.com/gist/jvishnuvardhan/23066f1a722fc566b437a5210d1b97b0/ssd_saved_model_tflite_conversion.ipynb) here.

Please verify once and close the issue if this was resolved for you. Thanks

**Comment by jvishnuvardhan** [comment]
@bgy-proprietor I converted yolo-v3 as shown in [this](https://github.com/jvishnuvardhan/TF_Lite/blob/master/TFlite_Yolov3_conversion.ipynb) gist. Try it , if it is not working for you then please create a new issue (with a standalone code) so that it will be easy for the community to follow. Thanks

**Comment by jvishnuvardhan** [comment]
@rgbarishian @mgalgs Can you please try `tf-nightly` and if it is not working then can you please create a separate issue with a standalone code to reproduce the issue? It will be easy for the community to follow. Thanks

**Comment by rgbarishian** [comment]
I found the error in my code. @Ringhu  may have the same error. 
When importing MobileNet, I did not give input_shape after setting include_top to False. After adding that kwarg, error no longer occurs

**Comment by mgalgs** [comment]
> @rgbarishian @mgalgs Can you please try `tf-nightly` and if it is not working then can you please create a separate issue with a standalone code to reproduce the issue? It will be easy for the community to follow. Thanks

Ok, just tested again and I'm not seeing this error with `tf-nightly` :+1:

**Comment by jvishnuvardhan** [comment]
Thanks for the confirmation. I will close this issue as this was resolved. Thanks!

**Comment by mgalgs** [comment]
@jvishnuvardhan so it turns out that I can't use `tf-nightly` to convert a model which was created using Tensorflow 1.15 (it dumps a huge error which also appears to dump the tensors themselves, 216MB worth of logs so not sharing...).  I'm guessing you have to generate the model using the same version of TF as you use to convert it?

Unfortunately, I can't generate my model using `tf-nightly` because I depend on on the [`object_detection` API](https://github.com/tensorflow/models/tree/master/research/object_detection), which isn't yet supported on Tensorflow 2...

So I'm stuck between TF 1 and 2... :sob: Any suggestions?

**Comment by hahmad2008** [comment]
> @Ringhu Sorry for the delay in my response. I tried converting the `mobilenet` model and was successful. Can you please check the [gist](https://colab.research.google.com/gist/jvishnuvardhan/23066f1a722fc566b437a5210d1b97b0/ssd_saved_model_tflite_conversion.ipynb) here.
> 
> Please verify once and close the issue if this was resolved for you. Thanks


Hi @jvishnuvardhan 
I tried to convert the checkpoint (saved_model) using this [solution](https://colab.research.google.com/gist/jvishnuvardhan/23066f1a722fc566b437a5210d1b97b0/ssd_saved_model_tflite_conversion.ipynb#scrollTo=X5s5D5ufCcKb), it is successfully converted but I think it is incorrect since I checked the input for the model I got it as [1,1,1,3] instead of [1,300,300,3]

the saved model directory has: 
-
![image](https://user-images.githubusercontent.com/68266028/89431979-40533200-d749-11ea-9cee-d492b567758b.png)

**Comment by agql** [comment]
Dear all,

Sorry to post on a closed issue but I'm experimenting this issue as well.

------------
I am trying to convert two models I trained to TFLITE format for using them in an Android app. Following the indications at https://www.tensorflow.org/lite/convert I get this mistake.

I have tried with a pre-trained model (ssd_resnet50_v1_fpn_640x640_coco17_tpu-8) and get the same error message
   _ValueError: None is only supported in the 1st dimension. Tensor 'input_tensor' has invalid shape '[1, None, None, 3]'._

I am running Windows 10 with Python 3.7.6 and Tensorflow 2.2
-----------
I have tried in a GCloud linux machine (Linux 4.9.0-12-amd64 #1 SMP Debian 4.9.210-1+deb9u1 (2020-06-07) x86_64 GNU/Linux) with Python 3.7.6 and Tensorflow 2.3.0 and a get:

_tensorflow.lite.python.convert.ConverterError: <unknown>:0: error: loc("Func/StatefulPartitionedCall/input/_0"): requires all operands and results to have compatible element types
<unknown>:0: note: loc("Func/StatefulPartitionedCall/input/_0"): see current operation: %1 = "tf.Identity"(%arg0) {device = ""} : (tensor<1x?x?x3x!tf.quint8>) -> tensor<1x?x?x3xui8>_

------------

Any help will be more than appreciated

Bests
